schema/loader.go

```diff
@@ -0,0 +1,159 @@
+package schema
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/santhosh-tekuri/jsonschema/v5"
+	"github.com/santhosh-tekuri/jsonschema/v5/httploader"
+)
+
+var (
+	//go:embed known_schemas
+	knownSchemas embed.FS
+)
+
+type (
+	SchemaFile string
+)
+
+const (
+	schemaDirectory = "known_schemas/"
+
+	// Presentation Exchange Schemas
+	PresentationDefinitionSchema              SchemaFile = "pe-presentation-definition.json"
+	PresentationDefinitionEnvelopeSchema      SchemaFile = "pe-presentation-definition-envelope.json"
+	PresentationSubmissionSchema              SchemaFile = "pe-presentation-submission.json"
+	PresentationClaimFormatDesignationsSchema SchemaFile = "pe-definition-claim-format-designations.json"
+	SubmissionClaimFormatDesignationsSchema   SchemaFile = "pe-submission-claim-format-designations.json"
+	SubmissionRequirementSchema               SchemaFile = "pe-submission-requirement.json"
+	SubmissionRequirementsSchema              SchemaFile = "pe-submission-requirements.json"
+
+	PresentationClaimFormatDesignationFormatDefinition SchemaFile = "pe-submission-claim-format-designations.json"
+
+	// Credential Manifest Schemas
+	CredentialManifestSchema    SchemaFile = "cm-credential-manifest.json"
+	CredentialApplicationSchema SchemaFile = "cm-credential-application.json"
+	CredentialResponseSchema    SchemaFile = "cm-credential-response.json"
+	OutputDescriptorsSchema     SchemaFile = "cm-output-descriptors.json"
+
+	// Wallet Rendering Schemas
+	DisplayMappingObjectSchema        SchemaFile = "wr-display-mapping-object.json"
+	EntityStylesSchema                SchemaFile = "wr-entity-styles.json"
+	LabeledDisplayMappingObjectSchema SchemaFile = "wr-labeled-display-mapping-object.json"
+
+	// VC JSON Schema Schemas
+	VerifiableCredentialJSONSchemaSchema SchemaFile = "vc-json-schema.json"
+)
+
+func (s SchemaFile) String() string {
+	return string(s)
+}
+
+// CachingLoader is a struct that holds local schemas
+type CachingLoader struct {
+	schemas sync.Map
+}
+
+// NewCachingLoader returns a new CachingLoader that enables the ability to cache http and https schemas
+func NewCachingLoader(schemas map[string]string) (*CachingLoader, error) {
+	// make sure only one process can write to the map at a time
+	cl := CachingLoader{schemas: sync.Map{}}
```

Comment:
nit: https://github.com/uber-go/guide/blob/master/style.md#omit-zero-value-fields-in-structs

```suggestion
	cl := CachingLoader{}
```